packages/typescript-estree/src/parseSettings/ExpiringCache.ts

```diff
@@ -0,0 +1,111 @@
+export const DEFAULT_TSCONFIG_CACHE_DURATION_SECONDS = 30;
```

Comment:
I think you might thinking in terms of the same glob being resolved sequentially and this cache short-circuiting that case.
That's not what this cache is for (though it will help with that case).

When we resolve the "parser settings", we resolve the globs to specific tsconfig paths on disk.
Currently every single time ESLint parses a file we do this glob resolution step.
For the persistent case this is "okay" because we treat the disk as unstable - i.e. we assume the globs could match something different (even though 99.999% of the time they don't).
For the single-run case this is BAD because we assume the disk is immutable - so we waste so much time re-resolving globs.

Setting this time too low means we'll resolve globs unnecessarily and not save much time. For example imagine you're typing in your IDE, then pause for 1s to think, then continue typing - that 1s was long enough to invalidate the glob resolution cache, so your next keystroke will be linted and we'll attempt to re-resolve the globs.

So why are we using 30s? I picked it because [eslint-plugin-import uses 30s for their cache](https://github.com/import-js/eslint-plugin-import/blob/766af5ff6c2e97cbe2a2a211729f7c129da3005d/utils/ModuleCache.js#L35-L37). But also it probably strikes a good balance between reactivity and performance in the persistent IDE usecase (and other non-single-run-but-are-actually-single-run cases).

Reply:
Got it, makes sense. 30s it is - thanks for the explanation!